discovery/discovery.go

```diff
@@ -26,17 +26,15 @@ func Dir() string {
 		return pluginDir
 	}
 
-	usr, err := user.Current()
-	if err != nil {
-		panic(err)
+	home := os.Getenv("HOME")
+	if usr, err := user.Current(); err == nil {
+		home = usr.HomeDir
 	}
-	return path.Join(usr.HomeDir, ".infrakit/plugins")
+	return path.Join(home, ".infrakit/plugins")
 }
 
 // NewPluginDiscovery creates a plugin discovery based on the environment configuration.
-func NewPluginDiscovery() (Plugins, error) {
-	pluginDir := Dir()
```

Comment:
By having the parameter in the function we are
  
  + being explicit about the requirements to construct a discovery object (the directory to scan).  Environment variable
  + following a pattern similar to IoC (inversion of control) or dependency injection -- meaning it's up to the caller to configure the object at construction time (`NewPluginDiscovery` is like a constructor in that it creates and returns a new instance).
  + some of the tests I am working on need to set this directory explicitly and the code is easier to follow with the second form, especially when the tests and os process boundaries aren't clear and tests may be executed concurrently.  

Having said that,  I am not crazy about all the changes in the main.go files.  Let me do this:

  1. Add a new function that allows setting of the directory
  2. Change the implementation of `NewPluginDiscovery() to call that new function with the `Dir()` function which returns a default location
  3. Revert the main.go files back to using the 'default constructor' form so that we have a simple, uniform default.

 @wfarner